####################################
InterPro Consortium Member databases
####################################

***********
CATH-Gene3D
***********

CATH-Gene3D database describes protein families and domain architectures in
complete genomes. Protein families are formed using a Markov clustering
algorithm, followed by multi-linkage clustering according to sequence identity.
Mapping of predicted structure and sequence domains is undertaken using hidden
Markov models libraries representing CATH and Pfam domains. CATH-Gene3D is based
at University College, London, UK.

***
CDD
***

CDD is a protein annotation resource that consists of a collection of annotated
multiple sequence alignment models for ancient domains and full-length proteins.
These are available as position-specific score matrices (PSSMs) for fast
identification of conserved domains in protein sequences via RPS-BLAST. CDD
content includes NCBI-curated domain models, which use 3D-structure information
to explicitly define domain boundaries and provide insights into
sequence/structure/function relationships, as well as domain models imported
from a number of external source databases.

*****
HAMAP
*****

HAMAP stands for High-quality Automated and Manual Annotation of Proteins. HAMAP
profiles are manually created by expert curators. They identify proteins that
are part of well-conserved proteins families or subfamilies. HAMAP is based at
the SIB Swiss Institute of Bioinformatics, Geneva, Switzerland.

***********
MobiDB Lite
***********

MobiDB offers a centralized resource for annotations of intrinsic protein
disorder. The database features three levels of annotation: manually curated,
indirect and predicted. The different sources present a clear tradeoff between
quality and coverage. By combining them all into a consensus annotation, MobiDB
aims at giving the best possible picture of the “disorder landscape” of a given
protein of interest.

*******
PANTHER
*******

PANTHER is a large collection of protein families that have been subdivided into
functionally related subfamilies, using human expertise. These subfamilies model
the divergence of specific functions within protein families, allowing more
accurate association with function, as well as inference of amino acids
important for functional specificity. Hidden Markov models (HMMs) are built for
each family and subfamily for classifying additional protein sequences. PANTHER
is based at at University of Southern California, CA, US.

****
Pfam
****

Pfam is a large collection of multiple sequence alignments and hidden Markov
models covering many common protein domains. Pfam is based at EMBL-EBI, Hinxton,
UK.

*****
PIRSF
*****

PIRSF protein classification system is a network with multiple levels of
sequence diversity from superfamilies to subfamilies that reflects the
evolutionary relationship of full-length proteins and domains. PIRSF is based at
the Protein Information Resource, Georgetown University Medical Centre,
Washington DC, US.

******
PRINTS
******

PRINTS is a compendium of protein fingerprints. A fingerprint is a group of
conserved motifs used to characterise a protein family or domain. PRINTS is
based at the University of Manchester, UK.

****************
PROSITE profiles
****************

PROSITE is a database of protein families and domains. It consists of
biologically significant sites, patterns and profiles that help to reliably
identify to which known protein family a new sequence belongs. PROSITE is base
at the Swiss Institute of Bioinformatics (SIB), Geneva, Switzerland.

****************
PROSITE patterns
****************

PROSITE is a database of protein families and domains. It consists of
biologically significant sites, patterns and profiles that help to reliably
identify to which known protein family a new sequence belongs. PROSITE is base
at the Swiss Institute of Bioinformatics (SIB), Geneva, Switzerland

****
SFLD
****

SFLD (Structure-Function Linkage Database) is a hierarchical classification of
enzymes that relates specific sequence-structure features to specific chemical
capabilities.

*****
SMART
*****

SMART (a Simple Modular Architecture Research Tool) allows the identification
and annotation of genetically mobile domains and the analysis of domain
architectures. SMART is based at at EMBL, Heidelberg, Germany.

***********
SUPERFAMILY
***********

SUPERFAMILY is a library of profile hidden Markov models that represent all
proteins of known structure. The library is based on the SCOP classification of
proteins: each model corresponds to a SCOP domain and aims to represent the
entire SCOP superfamily that the domain belongs to. SUPERFAMILY is based at the
University of Bristol, UK.

********
TIGRFAMs
********

TIGRFAMs is a collection of protein families, featuring curated multiple
sequence alignments, hidden Markov models (HMMs) and annotation, which provides
a tool for identifying functionally related proteins based on sequence homology.
TIGRFAMs is based at the J. Craig Venter Institute, Rockville, MD, US.
